include/aws_iot_mqtt_client.h

```diff
@@ -397,9 +395,9 @@ bool aws_iot_mqtt_is_client_connected(AWS_IoT_Client *pClient);
  *
  * @return The state of the MQTT client context at the time of the function call.
  *
- * @warning Application code should not rely on this function's return value.
- * The client's state may change between the time this function is called
- * and the application checks its return value.
+ * @note The client's state is internal and generally not useful to application code.
```

Comment:
I'm concerned that this change of text invokes the question of _why does the API exist_ if the documentation says that it is not  (generally) useful to the application.
The expectation of the reader can be for the documentation to mention when exactly this function is of use to the application.

Reply:
It's called internally in the SDK.  It probably should be an internal function, but is declared in this header and we should not change the contents of this header.